### PR TITLE
Add windows libffi.dll and remove old code

### PR DESCRIPTION
**Pull request by jyapayne**
These dlls were compiled with mingw. Libffi should now work with both windows 32 and 64 bit.

### CODE REVIEW

**Comment by jyapayne** [comment]
Note that the current tests in the Nim repo might need to be reworked for windows because when I run them, I get an error about stderr not being able to be imported.

File: libffi.nim [hunk 1]
```diff
@@ -11,36 +11,15 @@
 
 when defined(windows):
   import os
-  # on Windows we don't use a DLL but instead embed libffi directly:
```
Comment (Araq) [inline libffi.nim:14]: Oh sorry, I wasn't aware we do that. Isn't that much better than relying on DLLs?

Comment (jyapayne) [reply libffi.nim:14]: Yeah. I tried so hard to get compiling the 64 bit version to work. Something is wrong with the assembly sources and I have no idea what.

Comment (jyapayne) [reply libffi.nim:14]: Still trying to get a system that works for building the windows code. 64 bit code just fails and it's not clear why.

**Comment by jyapayne** [comment]
@Araq I'm afraid I'm out of my depth here. Libffi has a confusing make process and I'm not skilled enough with C, configure, and Makefiles to figure out what's going on. You can accept this PR or close it in favor of a better solution. I honestly prefer having the compilation vs DLLs.

**Comment by ringabout** [comment]
It's painful to install libffi on windows.

Without this PR, it is broken on 64-bit windows.

With this PR, I still got some errors:
```
nim c -d:release -d:nimHasLibFFI -o:nim_ffi2 compiler/nim.nim
nim_ffi2.exe c -r --experimental:compiletimeFFI tests/vm/mevalffi.nim 
```

```
Error: cannot open 'C:\Users\blue\Documents\GitHub\lib\system.nim'
```

**Comment by ringabout** [comment]
@Araq

Hi, Can this PR be merged?
I test it on windows with mingw64. It works like a charm.
```
koch temp -d:nimHasLibFFi
bin/nim_temp c -r --experimental:compiletimeFFI -d: nimEvalffiStderrWorkaround tests/vm/mevalffi.nim
```


```
MulCmp_352321690
foo
foo:100
foo:101
foo:102:103
foo:102:103:104
foo:0.03:asdf:103:105
ret=[s1:foobar s2:foobar age:25 pi:3.14]

foo
foo:100
foo:101
foo:102:103
foo:102:103:104
foo:0.03:asdf:103:105
ret=[s1:foobar s2:foobar age:25 pi:3.14]
```

BTW another language chooses libffi as repl implementation. 
https://github.com/crystal-lang/crystal/pull/10910

**Pull request merged**